src/devices/machine/mm5740.cpp

```diff
@@ -100,7 +97,12 @@ TIMER_CALLBACK_MEMBER(mm5740_device::perform_scan)
 
 	for (int x = 0; x < 9; x++)
 	{
-		uint16_t data = m_read_x[x]() ^ 0x3ff;
+		u16 data = m_read_x[x]() ^ 0x3ff;
```

Comment:
Pre-existing issue, but this shouldn’t be scanning all X lines on every active clock edge, it should only be scanning one line on each active clock edge.

Reply:
If we need to only do one X line scan per clock, this class will need to be refactored to handle debounce differently. Currently it's using the debounce capacitor value to determine the key bounce mask time, based on the H19 schematics, the capacitor is 5000 pF, and based on a linear slope shown in the datasheet, this would put the bounce mask time at 40 mSec. The effective clock for the chip is just 25 Hz, as I've mentioned in another comment on this PR - https://github.com/mamedev/mame/pull/11352#discussion_r1248874602. If we only scanned one line on each clock, that would be less than 3 scans per line per second.

Reply:
Due to the way MAME works, the emuated inputs are only updated once per emulated video frame anyway (every 16 ms at 60 Hz).  It drives up input latency when debounce is emulated on top of that (as is the case on a number of .  Is the debounce emulation even implemented realistically here?

Anyway, probably something to worry about another time.